### PR TITLE
Extend relative uri support to redirectStartPage

### LINKED ISSUE #2855: redirectStartPage is causing an infinite loop
## Library

- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.8.0`
- [ ] `@azure/msal-node@1.x.x`
- [x] `@azure/msal-react@1.0.0-alpha.1`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angular@2.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description

I'm trying to use redirectStartPage in a react application, and every time I use it the page gets stuck in a loop, so I'm wondering if I'm using it incorrectly. The app below is a simple example. If I go to 'another secure page' - /secure/more, then it works fine, as there's no redirectStartPage parameter. But if I go to /secure then it keeps redirecting back to /.

I've tried various options, redirecting to different places, and the effect is always the same. Once it's got into a loop then it keeps looping, even if I close the site and come back to it.

The 'xxx.request.origin' item in the localStorage never seems to get cleared. Is there something I should do to make sure it is cleared?

```jsx
import React from "react";
import * as msal from "@azure/msal-browser";
import { MsalAuthenticationTemplate, MsalProvider } from "@azure/msal-react";
import { Link, Router } from "@reach/router";

const Public = () => (
  <div>
    <h1>Public page</h1>
    <ul>
      <li>
        <Link to="/secure">Go to secure page</Link>
      </li>
      <li>
        <Link to="/secure/more">Go to another secure page</Link>
      </li>
    </ul>
  </div>
);

const Secure = ({ title }) => (
  <MsalAuthenticationTemplate
    interactionType={msal.InteractionType.Redirect}
    authenticationRequest={{
      redirectStartPage: "/",
    }}
  >
    <h1>{title}</h1>
    <Link to="/">Go to public page</Link>
  </MsalAuthenticationTemplate>
);

const SecureMore = ({ title }) => (
  <MsalAuthenticationTemplate
    interactionType={msal.InteractionType.Redirect}
  >
    <h1>{title}</h1>
    <Link to="/">Go to public page</Link>
  </MsalAuthenticationTemplate>
);

export const App = () => {
  const msalInstance = new msal.PublicClientApplication({
    auth: {
      clientId: "0a61c279-646b-4055-a5f1-1c3da7f70f18",
      navigateToLoginRequestUrl: true,
      redirectUri: "http://localhost:8000/",
      validateAuthority: false,
    },
    cache: {
      cacheLocation: "localStorage",
      storeAuthStateInCookie: true,
    },
    system: {},
  });

  return (
    <MsalProvider instance={msalInstance}>
      <Router>
        <Public path="/" />
        <Secure path="/secure" title="Secure Page" />
        <SecureMore path="/secure/more" title="Another Secure Page" />
      </Router>
    </MsalProvider>
  );
};
```

## Source

- [ ] Internal (Microsoft)
- [x] Customer request

### PR DESCRIPTION
**Pull request by tnorling**
This PR adds support for relative uris on `redirectStartPage` using the same API that was used to support this for `redirectUri` and `postLogoutRedirectUri`

Closes #2855

### CODE REVIEW

File: change/@azure-msal-browser-2ca47cd6-90a9-4988-b2c3-d0fbeacecdad.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jasonnutter) [inline change/@azure-msal-browser-2ca47cd6-90a9-4988-b2c3-d0fbeacecdad.json:2]: I would consider this more of an enhancement, as opposed to a new feature.

**Review by jasonnutter — APPROVED** [review]
Question about versioning, and curious if any docs should be updated? Otherwise looks good.

**Review by pkanher617 — APPROVED** [review]

File: change/@azure-msal-browser-2ca47cd6-90a9-4988-b2c3-d0fbeacecdad.json (hunk 1 above)
Comment (tnorling) [inline change/@azure-msal-browser-2ca47cd6-90a9-4988-b2c3-d0fbeacecdad.json:2]:
```suggestion
  "type": "patch",
```

Comment (tnorling) [reply change/@azure-msal-browser-2ca47cd6-90a9-4988-b2c3-d0fbeacecdad.json]: Changed.

**Comment by tnorling** [comment]
> Question about versioning, and curious if any docs should be updated? Otherwise looks good.

Looked for docs that needed updating, but we don't have request docs anymore besides the typedocs

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 2]
```diff
@@ -726,6 +726,15 @@ export abstract class ClientApplication {
         return UrlString.getAbsoluteUrl(postLogoutRedirectUri, BrowserUtils.getCurrentUri());
     }
 
+    /**
+     * Use to get the redirectStartPage either from request or use current window
+     * @param requestStartPage 
+     */
+    protected getRedirectStartPage(requestStartPage?: string): string {
+        const redirectStartPage = requestStartPage || BrowserUtils.getCurrentUri();
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:734]:
```suggestion
        const redirectStartPage = requestStartPage || window.location.href;
```

**Pull request merged**